### PR TITLE
[libuuid] Added cmake config and targets

### LINKED ISSUE #18732: [libuuid] Add CMake config & targets
****

**Is your feature request related to a problem? Please describe.**
You can't find_package(libuuid) at the moment, which is unexpected.

**Proposed solution**
libuuid should provide CMake .config and .targets

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by bobuc-havok**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #18732 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all previously supported, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
Since this is unofficial targets, could you add the prefix 'unofficial-' to the configs and targets?

File: ports/libuuid/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: libuuid
-Version: 1.0.3-7
+Version: 1.0.3-8
```
Comment (PhoebeHui) [inline ports/libuuid/CONTROL:2]:
```suggestion
Version: 1.0.3
Port-Version: 8
```

Comment (bucurb) [reply ports/libuuid/CONTROL]: Done. Converted CONTROL -> vcpkg.json in the process as well.

File: versions/l-/libuuid.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "bdadd106940f7c668a2891842c7745e6c7ce021c",
+      "version": "1.0.3",
+      "port-version": 8
+    },
+    {
+      "git-tree": "f334c99420e98969f329593b5cbb69986e94e740",
+      "version-string": "1.0.3-8",
+      "port-version": 0
+    },
```
Comment (PhoebeHui) [inline versions/l-/libuuid.json:12]:
```suggestion
```

File: ports/libuuid/CMakeLists.txt [hunk 3]
```diff
@@ -29,7 +29,35 @@ endif()
...
+    FILE unofficial-libuuid-targets.cmake
+    DESTINATION "${INSTALL_CONFIG_DIR}"
+)
+
+install(
+    FILES
+        "${PACKAGE_CONFIG_FILE}"
+    DESTINATION
+        "${INSTALL_CONFIG_DIR}"
+)
```
Comment (dg0yt) [inline ports/libuuid/CMakeLists.txt:63]: I miss the trailing newline.

File: ports/libuuid/unofficial-libuuid-config.cmake.in [hunk 4]
```diff
@@ -0,0 +1,6 @@
+@PACKAGE_INIT@
+
+if(NOT TARGET unofficial::UUID::UUID)
+    include(CMakeFindDependencyMacro)
```
Comment (dg0yt) [inline ports/libuuid/unofficial-libuuid-config.cmake.in:4]: AFAICT `CMakeFindDependencyMacro` isn't used here.

File: ports/libuuid/unofficial-libuuid-config.cmake.in [hunk 5]
```diff
@@ -0,0 +1,6 @@
+@PACKAGE_INIT@
+
+if(NOT TARGET unofficial::UUID::UUID)
+    include(CMakeFindDependencyMacro)
+    include("${CMAKE_CURRENT_LIST_DIR}/unofficial-libuuid-targets.cmake")
+endif()
```
Comment (PhoebeHui) [inline ports/libuuid/unofficial-libuuid-config.cmake.in:6]:
```suggestion
    include("${CMAKE_CURRENT_LIST_DIR}/unofficial-libuuid-targets.cmake")
```

**Review by PhoebeHui — APPROVED** [review]

**Comment by PhoebeHui** [comment]
The failure in pistache:x64-linux would be fixed by https://github.com/microsoft/vcpkg/pull/18994.

**Pull request merged**